### PR TITLE
Move mps data to CPU in util.to_numpy

### PR DESCRIPTION
**Pull request by adelevie**
Ostensibly solves https://github.com/skorch-dev/skorch/issues/883, though I am not very familiar with this codebase.

### CODE REVIEW

**Comment by BenjaminBossan** [comment]
Thank you for the bug report and providing a possible solution. This is probably hard to test, especially with the CI we have. Did you test it for your use case and did it solve the issue?

The M1 support from PyTorch still seems to be WIP (see their issue 77764 (not linking there to avoid spam)) but I guess there is no harm in adding support for this even without a proper way of testing, since the change seems innocuous enough.

**Comment by adelevie** [comment]
I can say that the `TypeError: can't convert mps:0 device type tensor to numpy.` error went away and the model did train with no errors.

I figured duck-typing `hasattr(X, 'is_mps')` was the safest way to work around `torch` versions that lack MPS support. Looks like it didn't break anything already in CI.

**Comment by BenjaminBossan** [comment]
> I can say that the `TypeError: can't convert mps:0 device type tensor to numpy.` error went away and the model did train with no errors.

Okay, that's great.

> I figured duck-typing `hasattr(X, 'is_mps')` was the safest way to work around `torch` versions that lack MPS support. Looks like it didn't break anything already in CI.

Yes, I believe that's the best way of doing it (short of PyTorch providing an official, backwards compatbile method to check it).

I don't believe there is much more that can be done here, so I'm willing to merge. I'll just wait a few days to see if @ottonemo or @thomasjpfan have anything to add.

File: skorch/utils.py [hunk 1]
```diff
@@ -165,6 +165,10 @@ def to_numpy(X):
     if X.is_cuda:
         X = X.cpu()
 
+    if hasattr(X, 'is_mps'):
+        if X.is_mps:
+            X = X.cpu()
```
Comment (thomasjpfan) [inline skorch/utils.py:170]: Nit:

```suggestion
    if hasattr(X, 'is_mps') and X.is_mps:
        X = X.cpu()
```

**Review by thomasjpfan — COMMENTED** [review]
Overall, I am okay with this PR.

I think we can add a quick test for `to_numpy` that only runs when `mps` is available. Here is a snippet to check if `mp` is available:

```python
hasattr(torch.backends, "mps") and torch.backends.mps.is_available()
```

**Comment by BenjaminBossan** [comment]
> I think we can add a quick test for `to_numpy` that only runs when `mps` is available

Yes, why not. It will probably not run for the time being but at least we have something if mps should become available on CI (or one of the skorch users with a MB runs the test locally). @adelevie would you be so kind as to add a test?

**Comment by BenjaminBossan** [comment]
Let's just add a new test function to the existing class. The function checks if mps is available, as suggested by Thomas, and skips if it isn't. Then create an mps tensor, run `to_numpy` on it, and check that it's a numpy array. That should be good enough.

**Comment by adelevie** [comment]
Sounds good, shouldn’t be too much trouble.

**Comment by adelevie** [comment]
Should be good to go. I squashed to single cleaner commit, hope that's ok.

**Review by thomasjpfan — APPROVED** [review]
Thank you for the update! I tested this on my m1 mac and the test runs and passes.

**Comment by BenjaminBossan** [comment]
Thanks @adelevie and @thomasjpfan.

> I squashed to single cleaner commit, hope that's ok.

Yeah, that's okay, but we would have squashed anyway, so it wasn't necessary.

**Pull request merged**